airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/read/JdbcPartitionsCreator.kt

```diff
@@ -192,7 +194,9 @@ class JdbcConcurrentPartitionsCreator<
         // Ensure that the cursor upper bound is known, if required.
         if (partition is JdbcCursorPartition<*>) {
             ensureCursorUpperBound()
-            if (streamState.cursorUpperBound?.isNull == true) {
+            if (
+                streamState.cursorUpperBound == null || streamState.cursorUpperBound?.isNull == true
```

Comment:
This is a bit strange, it seems technically, it is necessary to just check one of them?

Is it is null, both of them needs to be set correctly?

Reply:
a json node of type `NullNode` (represented in json as `key=null`) is not a java null.
the `.isNull` is a Jackson library function, not a language internal.
So the two states may be possible